testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java

```diff
@@ -193,6 +195,66 @@ public void testCreateSchema()
         assertUpdate("DROP SCHEMA " + schemaName);
     }
 
+    @Test
+    public void testRenameTable()
+    {
+        if (!hasBehavior(SUPPORTS_RENAME_TABLE)) {
+            // must reference an existing table in ALTER TABLE since name is validated before calling renameTable
+            assertQueryFails("ALTER TABLE nation RENAME TO yyyy", "This connector does not support renaming tables");
+            return;
+        }
+
+        if (!hasBehavior(SUPPORTS_CREATE_TABLE)) {
+            throw new AssertionError("Cannot test ALTER TABLE RENAME without CREATE TABLE, the test needs to be implemented in a connector-specific way");
+        }
+
+        String oldTable = "test_rename_" + randomTableSuffix();
+        String newTable = "test_rename_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + oldTable + " (a bigint, b double)");
+        assertUpdate("ALTER TABLE " + oldTable + " RENAME TO " + newTable);
+        assertThat(query("SHOW TABLES LIKE '" + oldTable + "'")).returnsEmptyResult();
+        assertThat(query("SELECT a, b FROM " + newTable))
+                .returnsEmptyResult();
+        assertUpdate("INSERT INTO " + newTable + " (a, b) VALUES (42, -38.5)", 1);
+        assertThat(query("SELECT CAST(a AS bigint), b FROM " + newTable))
+                .matches("VALUES (BIGINT '42', -385e-1)");
+        assertUpdate("DROP TABLE " + newTable);
+    }
+
+    @Test
+    public void testRenameTableAcrossSchemas()
+    {
+        if (!hasBehavior(SUPPORTS_RENAME_TABLE_ACROSS_SCHEMAS)) {
+            // must reference an existing table in ALTER TABLE since name is validated before calling renameTable
+            assertQueryFails("ALTER TABLE nation RENAME TO other_schema.yyyy", "This connector does not support renaming tables");
+            return;
+        }
+
+        if (!hasBehavior(SUPPORTS_CREATE_SCHEMA)) {
```

Comment:
This is pointless. You are failing the test because needed behavior is missing.  Surely `assertUpdate("CREATE SCHEMA " + schemaName);` will fail it as well.

Reply:
I was following a convention from other tests. I guess this is just to have a more actionable error message.

It would be better not to require `CREATE SCHEMA` support here and instead have more than one schema in the test db, but I don't know if we can enforce/rely on this being true.